src/daemon.py

```diff
@@ -0,0 +1,87 @@
+import os
+import time
+import pandas as pd
+from web3 import Web3
+from threading import Thread
+from sqlalchemy import create_engine
+from src.imbalances_script import RawTokenImbalances
+from src.config import CHAIN_RPC_ENDPOINTS, CHAIN_SLEEP_TIMES
+
+def get_web3_instance(chain_name):
+    return Web3(Web3.HTTPProvider(CHAIN_RPC_ENDPOINTS[chain_name]))
+
+def get_finalized_block_number(web3):
+    return web3.eth.block_number - 64
+
+def create_db_connection(chain_name):
+    """Helper function that creates a connection to the prod db."""
+    db_name = "mainnet" if chain_name == 'Ethereum' else "xdai"
```

Comment:
tbh, in order to have a cleaner format for the environment variables, i would instead propose we use one variable per chain, i.e.,

`ETHEREUM_PROD_DB_URL`
`GNOSIS_PROD_DB_URL`

etc